Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -149,21 +149,26 @@ protected BlockPuller(ConcurrentChain chain, IReadOnlyNodesCollection nodes, Pro
             };
         }
 
-        /// <inheritdoc />
-        public virtual void PushBlock(int length, Block block, CancellationToken token)
+        /// <summary>
+        /// Method called when a new block is downloaded and pushed to the puller.
+        /// <para>
+        /// This method is to be overridden by derived classes. In the base class it only logs the event.
+        /// </para>
+        /// </summary>
+        /// <param name="blockHash">Hash of the newly downloaded block.</param>
+        /// <param name="downloadedBlock">Desciption of the newly downloaded block.</param>
+        /// <param name="cancellationToken">Cancellation token to be used by derived classes that allows the caller to cancel the execution of the operation.</param>
+        public virtual void BlockPushed(uint256 blockHash, DownloadedBlock downloadedBlock, CancellationToken cancellationToken)
         {
-            uint256 hash = block.Header.GetHash();
-
-            DownloadedBlock downloadedBlock = new DownloadedBlock()
-            {
-                Block = block,
-                Length = length,
-            };
+            this.logger.LogTrace($"(blockHash:'{blockHash}',downloadedBlock.Length:{downloadedBlock.Length})");
+            this.logger.LogTrace("(-)");
```

Comment:
That's very important thing of my logging style, but hard to explain here. Basically you want all entries and exists from the method to be in the log. This should be more evident when I fully log these as a part of quality score issue. Please ignore now.